enhancements/ingress/lb-allowed-source-ranges.md

```diff
@@ -0,0 +1,261 @@
+
+### Implementation Details/Notes/Constraints [optional]
+
+The value set for `AllowedSourceRanges` will be used to set `.Spec.LoadBalancerSourceRanges` 
+of the LoadBalancer Service in front of the router in cloud environments.
+
+### Risks and Mitigations
+
+One risk of this proposal is that customers may have already set the
+`.Spec.LoadBalancerSourceRanges` field of the LoadBalancer Service in a previous version. 
+This risk is mitigated by marking the cluster as non-upgradable in such case.
+
+### Drawbacks
+
+This enhancement may bring additional complexity for the customers who are 
+used to modify directly the `Spec.LoadBalancerSourceRanges` of the LoadBalancer 
+Service for the same purposes. 
+
+## Design Details
+
+### Open Questions [optional]
+
+1. On Kubernetes API reference, the field 
+[`loadBalancerSourceRanges`](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.24/#service-v1-core) 
+has the warning "This field will be ignored if the cloud-provider does not support the feature.".
+However, there is no information about which platforms support this. Is there any platform
+OpenShift runs on which does not support this feature?
+
+A. I could only find this statement from Kubernetes
+[v1.18 website repository](https://github.com/kubernetes/website/blob/dev-1.18/content/en/docs/tasks/access-application-cluster/configure-cloud-provider-firewall.md): 
+"This feature is currently supported on Google Compute Engine, Google Kubernetes Engine, 
+AWS Elastic Kubernetes Service, Azure Kubernetes Service, and IBM Cloud Kubernetes Service". 
+The current version of the document does not mention `.Spec.LoadBalancerSourceRanges` anymore.
+
+### Test Plan
+
+The operator that manages the IngressController Deployment and related
+resources has unit test coverage; for this enhancement, the unit tests are
+expanded to cover the additional functionality.  
+
+The operator has end-to-end tests; for this enhancement, a test is added that verifies
+the following scenarios:
+
+1. Adding/deleting/changing `AllowedSourceRanges` on the IngressController and observing `.Spec.LoadBalancerSourceRanges` on the LoadBalancer Service.
+
+2. Adding `service.beta.kubernetes.io/load-balancer-source-ranges` annotation to a LoadBalancer Service and observing cluster's upgradable status.
+
+There is an upgrade test in openshift/origin to make sure that the migration workflow explained under **Variation** above works seamlessly.
+
+### Graduation Criteria
+
+N/A
+
+#### Dev Preview -> Tech Preview
+
+N/A. This feature will go directly to GA.
+
+#### Tech Preview -> GA
+
+N/A. This feature will go directly to GA.
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+On upgrade, `AllowedSourceRanges` is omitted by default, which is consistent with the
+feature's absence in older versions.  On downgrade, the IngressController
+Deployment is updated, and `AllowedSourceRanges` is automatically
+removed from the Deployment.
```

Comment:
This is still a little confusing, especially the description of downgrades.  The `AllowedSourceRanges` field is empty by default, so it will be empty on an upgraded cluster, until the cluster admin sets it after the upgrade.  If the cluster admin sets `AllowedSourceRanges` and then downgrades the cluster, then the `AllowedSourceRanges` field will vanish from the CRD; however, the downgraded operator would *not* update or remove the `Spec.LoadBalancerSourceRanges` field value and would *not* update or remove the `service.beta.kubernetes.io/load-balancer-source-ranges` annotation.  In any case, none of this (not `AllowedSourceRanges` on the IngressController, nor `LoadBalancerSourceRanges` or `service.beta.kubernetes.io/load-balancer-source-ranges` on the service) should have any effect on any deployment.

Reply:
Thanks a lot, it is very well explained. I'm adding to the EP.